pkg/startup/startup.go

```diff
@@ -720,6 +750,24 @@ func configureIPPools(ctx context.Context, client client.Interface) {
 		return
 	}
 
+	// If CIDRs weren't specified through the environment variables, check if they're present in kubeadm's
+	// config map.
+	if (len(ipv4Pool) == 0 || len(ipv6Pool) == 0) && kubeadmConfig != nil {
+		v4, v6, err := extractKubeadmCIDRs(kubeadmConfig)
+		if err == nil {
+			if len(ipv4Pool) == 0 {
+				ipv4Pool = v4
+				log.Infof("found v4=%s in the kubeadm config map", ipv4Pool)
+			}
+			if len(ipv6Pool) == 0 {
+				ipv6Pool = v6
+				log.Infof("found v6=%s in the kubeadm config map", ipv6Pool)
+			}
+		} else {
+			log.WithError(err).Error("Failed to extract CIDRs from kubeadm config.")
```

Comment:
If this is actually an error, then I think we need to exit. If this is not exit-worthy, then I think we should downgrade this log message to a warning level log.

(We try to keep error level logs to things that are actually problems the user needs to action. I'm not sure if this is one of those?)